src/ui/settings/ViewAdvancedDNSSettings.qml

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.5
+import QtQuick.Controls 2.14
+import QtGraphicalEffects 1.14
+import QtQuick.Layouts 1.14
+import Mozilla.VPN 1.0
+import "../components"
+import "../themes/themes.js" as Theme
+
+import org.mozilla.Glean 0.15
+import telemetry 0.15
+
+
+Item {
+    id: root
+
+    VPNMenu {
+        id: menu
+        objectName: "settingsAdvancedDNSSettingsBackButton"
+
+        //% "Advanced DNS Settings"
+        title: qsTrId("vpn.settings.advancedDNSSettings.title")
+        isSettingsView: true
+    }
+
+    VPNFlickable {
+        id: vpnFlickable
+        property bool vpnIsOff: (VPNController.state === VPNController.StateOff)
+
+        anchors.top: menu.bottom
+        anchors.right: parent.right
+        anchors.left: parent.left
+        height: root.height - menu.height
+        flickContentHeight: col.childrenRect.height
+        interactive: flickContentHeight > height
+
+        Component.onCompleted: {
+            Glean.sample.dnsSettingsViewOpened.record();
+        }
+
+        Column {
+            id: col
+            width: parent.width
+            anchors.top: parent.top
+            anchors.topMargin: 18
+            spacing: Theme.windowMargin
+
+            // TODO: this should be a radiobox
+            VPNCheckBoxRow {
+                id: useGatewayDNS
+                objectName: "gatewayDNSEnabled"
+                width: parent.width - Theme.windowMargin
+
+
+                //% "Use default DNS"
+                labelText: qsTrId("vpn.advancedDNSSettings.gateway")
+                //% "Automatically use Mozilla VPN-protected DNS"
+                subLabelText: qsTrId("vpn.advancedDNSSettings.gateway.description")
+                isChecked: VPNSettings.useGatewayDNS
+                onClicked: {
+                    if (vpnFlickable.vpnIsOff) {
+                        VPNSettings.useGatewayDNS = true
+                    }
+                }
+            }
+
+            // TODO: this should be a radiobox
+            VPNCheckBoxRow {
+                id: useLocalDNS
+                objectName: "settingMozillaDNSEnabled"
+                width: parent.width - Theme.windowMargin
+
+                //% "Use local DNS"
+                labelText: qsTrId("vpn.advancedDNSSettings.localDNS")
+                //% "Resolve websites using a DNS in your local network"
```

Comment:
I feel like "Resolve websites" is imprecise enough to trigger someone who knows what a DNS does, and to not help someone who doesn't.

Reply:
@chrisrooneys what are your thoughts here?

Reply:
@flodolo @lesleyjanenorton is "Access" technically accurate enough?

Reply:
It isn't. The accurate version would be `Resolve domain names using a DNS…`, or `Resolve website domain names using a DNS…` if we want to keep mentionining "websites".

Reply:
Between those I think "Resolve website domain names using a DNS" covers both groups of users better.

Reply:
thank you @flodolo & @chrisrooneys !